src/components/map/Map.tsx

```diff
@@ -50,61 +69,69 @@ const MapIcon = ({
   return <MarkerIcon color={iconColor} size={iconSize} />;
 };
 
-type TCluster = {
-  clusterColor: string;
-  geometry: { coordinates: [number, number] };
-  id: string;
-  onPress: () => void;
-  properties: { point_count: number };
-};
-
 const renderCluster = (cluster: TCluster) => {
   const { clusterColor: backgroundColor, geometry, id, onPress, properties = {} } = cluster;
-  const { point_count: points } = properties;
+  const { point_count } = properties;
 
   return (
-    <Marker
+    <PointAnnotation
       key={`cluster-${id}`}
-      coordinate={{
-        longitude: geometry.coordinates[0],
-        latitude: geometry.coordinates[1]
-      }}
-      style={styles.clusterMarker}
-      onPress={onPress}
+      id={`cluster-${id}`}
+      coordinate={geometry.coordinates}
+      onSelected={onPress}
     >
-      {CIRCLE_SIZES.map((size, index) => (
-        <View
-          key={`circle-${index}`}
-          style={[
-            styles.clusterCircle,
-            {
-              backgroundColor,
-              borderRadius: normalize(size / 2),
-              height: normalize(size),
-              opacity: 0.2 * (index + 1),
-              width: normalize(size)
-            }
-          ]}
-        >
-          <RegularText lightest center smallest>
-            {points}
-          </RegularText>
-        </View>
-      ))}
-    </Marker>
+      <View style={styles.hitBox}>
+        {CIRCLE_SIZES.map((size, index) => (
+          <View
+            key={`circle-${index}`}
+            style={[
+              styles.clusterCircle,
+              {
+                backgroundColor,
+                borderRadius: normalize(size / 2),
+                height: normalize(size),
+                opacity: 0.2 * (index + 1),
+                width: normalize(size)
+              }
+            ]}
+          >
+            <RegularText lightest center smallest>
+              {point_count}
+            </RegularText>
+          </View>
+        ))}
+      </View>
+    </PointAnnotation>
   );
 };
 
 /* eslint-disable complexity */
```

Comment:
should stay in front of `export const Map = ({`

Reply:
With the new commit this file has been freed from complexity. So I will remove this line. 👍

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/pull/1154/commits/800969674da23ee9825df39f73ddb943008dbe34